src/main/java/appeng/tile/AEBaseInvTile.java

```diff
@@ -19,223 +19,119 @@
 package appeng.tile;
 
 
-import java.util.EnumMap;
+import java.util.List;
 
+import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
-import net.minecraft.block.Block;
-import net.minecraft.entity.player.EntityPlayer;
-import net.minecraft.inventory.IInventory;
-import net.minecraft.inventory.ISidedInventory;
 import net.minecraft.item.ItemStack;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.text.ITextComponent;
 import net.minecraft.util.text.TextComponentString;
 import net.minecraft.util.text.TextComponentTranslation;
+import net.minecraft.world.World;
 import net.minecraftforge.common.capabilities.Capability;
 import net.minecraftforge.items.CapabilityItemHandler;
 import net.minecraftforge.items.IItemHandler;
-import net.minecraftforge.items.wrapper.InvWrapper;
-import net.minecraftforge.items.wrapper.SidedInvWrapper;
+import net.minecraftforge.items.wrapper.EmptyHandler;
 
-import appeng.block.AEBaseBlock;
 import appeng.tile.events.TileEventType;
-import appeng.tile.inventory.IAEAppEngInventory;
-import appeng.tile.inventory.InvOperation;
+import appeng.util.helpers.ItemHandlerUtil;
+import appeng.util.inv.IAEAppEngInventory;
+import appeng.util.inv.InvOperation;
 
 
-public abstract class AEBaseInvTile extends AEBaseTile implements ISidedInventory, IAEAppEngInventory
+public abstract class AEBaseInvTile extends AEBaseTile implements IAEAppEngInventory
 {
 
-	private EnumMap<EnumFacing, IItemHandler> sidedItemHandler = new EnumMap<>( EnumFacing.class );
-
-	private IItemHandler itemHandler;
-
-	@Override
-	public String getName()
-	{
-		return this.getCustomInventoryName();
-	}
-
 	@TileEvent( TileEventType.WORLD_NBT_READ )
 	public void readFromNBT_AEBaseInvTile( final net.minecraft.nbt.NBTTagCompound data )
 	{
-		final IInventory inv = this.getInternalInventory();
-		final NBTTagCompound opt = data.getCompoundTag( "inv" );
-		for( int x = 0; x < inv.getSizeInventory(); x++ )
+		final IItemHandler inv = this.getInternalInventory();
+		if( inv != null )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2589)